### PR TITLE
refactor: restructure commit-generation config to [commit.generation]

### PR DESCRIPTION
**Pull request by max-sixty**
## Summary

- Restructures config from `[commit-generation]` to `[commit.generation]` for consistency with other nested config sections
- Replaces `command` + `args` with single `command` string (shell execution via `sh -c`)
- Adds deprecation warnings and automatic migration file generation for the old format
- Simplifies deprecation detection code (removes unused `has_args` tracking)
- Moves deprecated section filtering to callers for cleaner architecture

## Config format

```toml
# Old (deprecated, still works)
[commit-generation]
command = "llm"
args = ["-m", "claude-haiku-4.5"]

# New
[commit.generation]
command = "llm -m claude-haiku-4.5"
```

## Test plan

- [x] All tests pass on Ubuntu, macOS, and Windows
- [x] Unit tests cover deprecation detection, migration, and edge cases
- [x] Integration tests verify `wt config show` with deprecated config formats
- [x] Migration file generation and warning messages verified

🤖 Generated with [Claude Code](https://claude.ai/code)

Co-Authored-By: Claude <noreply@anthropic.com>

### CODE REVIEW

File: src/llm.rs [hunk 1]
```diff
@@ -1,14 +1,45 @@
...
+const SHELL_METACHARACTERS: &[char] = &[
+    '&', '|', ';', '<', '>', '$', '`', '\'', '"', '(', ')', '{', '}', '*', '?', '[', ']', '~', '!',
+    '\\',
+];
+
+/// Format a reproduction command, only wrapping with `sh -c` if needed.
+///
+/// Simple commands like `llm -m haiku` are shown as-is.
+/// Complex commands with shell syntax are wrapped: `sh -c 'complex && command'`
+fn format_reproduction_command(base_cmd: &str, llm_command: &str) -> String {
```
Comment (max-sixty) [inline src/llm.rs:25]: @claude wouldn't it be easier to just do a single format? why have both?

Comment (max-sixty) [reply src/llm.rs:25]: do you see the line that I'm commenting on? my comment is very specific to tuos area of code

**Comment by max-sixty** [comment]
closes #801
